src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -17,24 +17,31 @@ public class AuthenticationServiceImpl implements AuthenticationService {
 
     @Override
     public User login(String email, String password) throws AuthenticationException {
-        Optional<User> userFromDbOptional = userService.findByEmail(email);
-        if (userFromDbOptional.isPresent() && userFromDbOptional.get().getPassword()
-                .equals(HashUtil.hashPassword(password, userFromDbOptional.get().getSalt()))) {
-            return userFromDbOptional.get();
+        Optional<User> userOptional = userService.findByEmail(email);
+        if (userOptional.isPresent() && checkPassword(userOptional, password)) {
+            return userOptional.get();
         }
         throw new AuthenticationException("Couldn't authenticate user. "
                 + "Email or password is invalid");
     }
 
     @Override
     public User register(String email, String password) throws RegistrationException {
-        User user = new User();
-        user.setEmail(email);
-        user.setPassword(password);
+        if (password.isEmpty()) {
+            throw new RegistrationException("Password couldn't be empty");
+        }
         if (userService.findByEmail(email).isEmpty() && !password.isEmpty()) {
+            User user = new User();
+            user.setEmail(email);
+            user.setPassword(password);
             return userService.add(user);
         }
         throw new RegistrationException("User with the same email: " + email
                 + " already registered");
     }
+
+    private static boolean checkPassword(Optional<User> userOptional, String password) {
```

Comment:
why **static**?
```suggestion
    private static boolean checkPassword(User user, String password) {
```

Reply:
Static type sets automatically when I was creating a method. I'll already fix it.

Reply:
you have to understand when to use static